lib/validate.js

```diff
@@ -19,20 +19,14 @@ var ajv = require('ajv')({
 });
 //meta: false, // optional, to prevent adding draft-06 meta-schema
 
-var ajvFormats = require('ajv/lib/compile/formats.js');
-ajv.addFormat('uriref', ajvFormats.full['uri-reference']);
-ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
-ajv._refs['http://json-schema.org/schema'] = 'http://json-schema.org/draft-04/schema'; // optional, using unversioned URI is out of spec
-var metaSchema = require('ajv/lib/refs/json-schema-v5.json');
-ajv.addMetaSchema(metaSchema);
-ajv._opts.defaultMeta = metaSchema.id;
-
 const common = require('./common.js');
 const jptr = require('reftools/lib/jptr.js');
 const walkSchema = require('./walkSchema.js').walkSchema;
 const wsGetDefaultState = require('./walkSchema.js').getDefaultState;
 const linter = require('./linter.js');
 
+setupAjv(ajv);
```

Comment:
Why call `setupAjv` before the rest of the constants/variables? Not a problem, just curious.

Reply:
It’s used two lines below to set up other constants. This whole things a bit of a mess but it’s sliiiightly better like this.